src/Catalog/Helpers/GalleryDatabaseQueryService.cs

```diff
@@ -223,13 +223,49 @@ private async Task<IReadOnlyCollection<FeedPackageDetails>> GetPackages(Db2Catal
 
                 using (_telemetryService.TrackGetPackageDetailsQueryDuration(cursor))
                 {
-                    using (var packagesReader = await packagesCommand.ExecuteReaderAsync())
+                    return await ReadPackages(packagesCommand);
+                }
+            }
+        }
+
+        private async Task<IReadOnlyCollection<FeedPackageDetails>> ReadPackages(SqlCommand packagesCommand)
```

Comment:
Nit: our style is to always add an `Async` suffix to async methods.

```suggestion
        private async Task<IReadOnlyCollection<FeedPackageDetails>> ReadPackagesAsync(SqlCommand packagesCommand)
```